File: 4_multiple_inheritance.cpp
#include<iostream>
#include<bits/stdc++.h>
using namespace std;

class Food{
    public:
    int cost;
    string type;
    Food(int cost, string type){
        this->cost=cost;
        this->type= type;
    }
    void display(){
        cout<<"Food is necessary"<<endl;
    }

};

class Product{
    public:
    // int amount;
    // string type;
    // Product(int amount, string type){
    //     this->amount=amount;
    //     this->type= type;
    // }

    void show(){
        cout<<"Products are to be sold"<<endl;
    }

};

class Dessert: public Food, public Product{
    public:
    Dessert(int cost, string type): Food(cost, type){

    }
};

int main(){
    Dessert cake(20, "Sweet");
    cake.display();
    cake.show();
   
return 0;
}

File: 4_multiple_inheritance.py
class Food:
    cost= None
    type= None
    def __init__(self, cost, type) -> None:
        self.cost= cost
        self.type= type
        
    def __str__(self) -> str:
        return f"{self.type} type of food costs {self.cost} BDT"
    
class Product:
    amount= None
    
    def display(self):
        print("Products need to be sold")
        
class Dessert(Food, Product):
    pass

cake= Dessert(20, "Sweet")
cake.display()
print(cake)

# Output 
# Products need to be sold
# Sweet type of food costs 20 BDT